### DISCUSSION TITLE
Keep selected node alive after deleting the contents of the search bar in "Create new node"

### DISCUSSION THREAD

**Discussion by PereViader** [godotengine/godot-proposals#7795] [Editor]
In Godot v4.1.1 if you search for a node in the "Scene", then select it and after that delete the search string, the node is still selected.

However, if you do the same in the "Create New Node" window, the selected node is lost and the root type "Node" is selected instead.

Keeping the selection in the "Create New Node" alive would be very useful, just like it is in the "Scene". You might remember the name of some node but not another, and with this you could look for a sibling and keep searching from there.

**Comment by aXu-AP** [comment]
Yes, I also often have this problem.
The main problem here is that modifying the search string triggers the search on each deleted letter, possibly preferring another result. So it would make sense to run the search upon search string getting shorter, but keep the last result selected. In the end, the selected node should still match the text (unless you are deleting letters from the center).

**Comment by Calinou** [comment]
I can reproduce this on 4.2.dev5.

Fixing this may not be trivial, as the most suited node is automatically selected when the search textr changes in any way. You can only have one node selected at a time in the Create New Node dialog, which I believe also impacts how this could be fixed.

https://github.com/godotengine/godot-proposals/assets/180032/568291f7-0ac9-42ab-bf8b-384601db6848

**Reply by PereViader** [reply]
I think it makes sense to select the most appropriate node given the search string as it makes selection much faster as well.

Maybe the suggested behaviour could make sense when the search string is completely deleted by pressing the X button on the right.

Alternatively, the node selection routine could check if the search string has had an additive change or subtractive change and only update the selected node on additive operations.
